jax/__init__.py

```diff
@@ -86,6 +86,7 @@
 # jax and rely on the names imported above.
 from . import image
 from . import lax
+from . import lax_reference
```

Comment:
I'd be reluctant to make `lax_reference` a public API.

Reply:
@hawkinsp We need this import in order to populate the `numpy_impl` rules. Do you have an alternative in mind?

Reply:
Perhaps the Numpy impl rules should live in either (a) the same module as the numpy evaluator or (b) the same module in which the primitives are defined? That way if you import `numpy_eval`, you will have the rules also.

Reply:
Given that this is a backend, I think we should keep it out of `lax.py`, similar to the TF backend etc. I'll try moving it into `numpy_rules` and see how it goes, wdyt?